libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/RouteArrowComponent.kt

```diff
@@ -23,17 +25,25 @@ class RouteArrowComponent(
         super.onAttached(mapboxNavigation)
         coroutineScope.launch {
             mapboxNavigation.flowRouteProgress().collect { routeProgress ->
-                mapView.getMapboxMap().getStyle()?.also { style ->
-                    val arrowUpdate = routeArrowApi.addUpcomingManeuverArrow(routeProgress)
+                val arrowUpdate = routeArrowApi.addUpcomingManeuverArrow(routeProgress)
+                mapboxMap.getStyle()?.let { style ->
                     routeArrowView.renderManeuverUpdate(style, arrowUpdate)
                 }
             }
         }
+
+        coroutineScope.launch {
+            mapboxNavigation.flowRoutesUpdated().filter { it.navigationRoutes.isEmpty() }.collect {
```

Comment:
I'm not completely sure what you're trying to do here, but I just want to point out that if you set empty routes when there are no routes present at the moment, you won't get an update. Also you won't get an update if the observer is registered after the empty routes are set. More info here: https://github.com/mapbox/mapbox-navigation-android/pull/6052
Probably that would not be an issue here, just pointing out.

Reply:
Thanks for sharing am thinking more 🤔 . This code is not new, it is from the [CarRouteLine](https://github.com/mapbox/mapbox-navigation-android/blob/152a4d5ee1619e6b9588ba3bae18847fbb804b7a/libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt#L75-L76). Here is the pull request that fixed an issue https://github.com/mapbox/1tap-android/pull/1206

I think you're referring to this [guarding condition in MapboxDirectionsSession](https://github.com/mapbox/mapbox-navigation-android/blob/152a4d5ee1619e6b9588ba3bae18847fbb804b7a/libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt#L47-L49). Meaning, `flowRoutesUpdated` will only emit empty when routes transition from non-empty to empty.

This is indeed, the only case we need this flowable to emit. It does not need to emit on empty to empty transitions, and would not be harmful if it did but it would be a no-op. 

So this all looks fine, I also spent some time trying to write a test that could produce an issue but did not find one.

Reply:
Maybe I'm mistaken considering there are some instrumentation test issues

![Screen Shot 2022-07-14 at 10 38 18 AM](https://user-images.githubusercontent.com/3021882/179047471-bb5c4efd-1562-4ad6-8e77-677b60399f91.png)

Reply:
Nevermind those are on the `main` line and have nothing to do with this code